library/blivet.py

```diff
@@ -555,14 +555,35 @@ def _create(self):
             raise BlivetAnsibleError("invalid size '%s' specified for volume '%s'" % (self._volume['size'], self._volume['name']))
 
         fmt = self._get_format()
-        if size > parent.free_space:
+
+        create_vdo = self._blivet_pool._pool['deduplication'] or self._blivet_pool._pool['compression']
+
+        if create_vdo:
+            pool_size = size
+            if self._blivet_pool._pool['vdo_pool_size']:
+                pool_size = Size(self._blivet_pool._pool['vdo_pool_size'])
+                if pool_size > parent.free_space:
+                    raise BlivetAnsibleError("specified 'vdo_pool_size' for volume '%s' "
+                                             "exceeds available space in pool '%s' (%s)" % (pool_size,
+                                                                                            parent.name,
+                                                                                            parent.free_space))
+
+            try:
+                vdopool = self._blivet.new_lv(name='vdopool', vdo_pool=True,
+                                            parents=[parent], compression=self._blivet_pool._pool['compression'],
+                                            deduplication=self._blivet_pool._pool['deduplication'],
+                                            size=pool_size)
+            except Exception as e:
+                raise BlivetAnsibleError("failed to set up VDO pool '%s': %s" % (self._volume['name'], str(e)))
+            self._blivet.create_device(vdopool)
+        elif size > parent.free_space:
             raise BlivetAnsibleError("specified size for volume '%s' exceeds available space in pool '%s' (%s)" % (size,
                                                                                                                    parent.name,
                                                                                                                    parent.free_space))
-
         try:
-            device = self._blivet.new_lv(name=self._volume['name'],
-                                         parents=[parent], size=size, fmt=fmt)
+            device = self._blivet.new_lv(name=self._volume['name'], vdo_lv=create_vdo,
```

Comment:
getting this error from ci testing: https://fedorapeople.org/groups/linuxsystemroles/logs/linux-system-roles-storage-pull-linux-system-roles_storage-181-638463f-centos-7-20210210-141946/artifacts/ansible.log
```
TASK [linux-system-roles.storage : manage the pools and volumes to match the specified state] ***
task path: /tmp/tmp0iw25j48/tasks/main-blivet.yml:104
fatal: [/cache/centos-7.qcow2c]: FAILED! => {"actions": [], "changed": false, "crypts": [], "leaves": [], "mounts": [], "msg": "failed to set up volume 'test1': __init__() got an unexpected keyword argument 'vdo_lv'", "packages": [], "pools": [], "volumes": []}
```
do you need to use a newer version of `blivet` that has support for the `vdo_lv` parameter?

Reply:
Yes, this is probable cause. However, the tests have passed on this before. So unless there was some change in CI configuration during that time, something went wrong with the change I have commited.